src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -290,9 +395,123 @@ public WindowTownHall(final BuildingTownHall.View townHall)
         registerButton(BUTTON_CHANGE_SPEC, this::doNothing);
         registerButton(BUTTON_TOGGLE_JOB, this::toggleHiring);
 
+        registerButton(BUTTON_PREV_PAGE_PERM, this::switchPage);
+        registerButton(BUTTON_NEXT_PAGE_PERM, this::switchPage);
+
+        registerButton(BUTTON_MANAGE_OFFICER, this::editOfficer);
+        registerButton(BUTTON_MANAGE_FRIEND, this::editFriend);
+        registerButton(BUTTON_MANAGE_NEUTRAL, this::editNeutral);
+        registerButton(BUTTON_MANAGE_HOSTILE, this::editHostile);
+
+
         registerButton(BUTTON_UP, this::updatePriority);
         registerButton(BUTTON_DOWN, this::updatePriority);
         registerButton(BUTTON_DELETE, this::deleteWorkOrder);
+
+        findPaneOfTypeByID(BUTTON_PREV_PAGE_PERM, Button.class).setEnabled(false);
+        findPaneOfTypeByID(BUTTON_MANAGE_OFFICER, Button.class).setEnabled(false);
+        //todo set up al the buttons with messages for the permission changes!
+
+        registerButton(BUTTON_TRIGGER, this::trigger);
+
+    }
+
+    public void trigger(Button button)
+    {
+        @NotNull final Pane pane = button.getParent().getChildren().get(2);
+        int index = 0;
+        if(pane instanceof Label)
+        {
+            index = Integer.valueOf(((Label) pane).getLabelText());
+        }
+        final boolean trigger = !(LanguageHandler.format("com.minecolonies.coremod.gui.workerHuts.retrieveOn")).equals(button.getLabel());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Define a constant instead of duplicating this literal "com.minecolonies.coremod.gui.workerHuts.retrieveOn" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)